app/models/concerns/orchestration/templates.rb

```diff
@@ -0,0 +1,33 @@
+module Orchestration::Templates
+  extend ActiveSupport::Concern
+  include Orchestration::Common
+
+  included do
+    after_validation :queue_render_checks
+  end
+
+  def queue_render_checks
+    return if skip_orchestration?
+
+    logger.debug "Scheduling render checks of provisioning templates for #{self}"
+
+    templates = [template_for_pxe_loader, provisioning_template].compact
```

Comment:
Shoudn't it be 
```suggestion
    templates = [template_for_pxe_loader, available_template_kinds(provisioning).values].compact
```
We're not only looking for a provisioning template. We are looking for the whole set of templates that are relevant to this host and its provisioning type. BTW the provisioning kind is kept as a separate parameter in the controller: https://github.com/theforeman/foreman/blob/c338bc83b002c92c4efb95724a9bb86bda47fcc9/app/controllers/hosts_controller.rb#L620
Make sure you're passing the kind properly from the UI

Reply:
> We are looking for the whole set of templates that are relevant to this host and its provisioning type. 

Do we? I want to check templates only relevant to the actual (selected) provisioning method. I don't want to check all template kinds assigned to the host.

Reply:
The `available_template_kinds` returns templates for all template kinds assigned to the host, unless it's in image mode, [see here](https://github.com/theforeman/foreman/blob/develop/app/models/concerns/hostext/operating_system.rb#L34).

That's what I don't want to do. I don't care about the render status of the iPXE template when I don't do iPXE provisioning.

Reply:
While it's better than nothing, can we test more templates? 
For example in image mode, we need to render more than one template, also in PXE use cases you have the pxe template that invokes the installer and then it is changed to the local disk template.

Reply:
The problem is determining which templates to check for every possible provisioning combination correctly.

Reply:
We always render all TFTP methods:
https://github.com/theforeman/foreman/blob/a3962dfdc385c9a3786062ae404b325cad8e89e4/app/models/concerns/orchestration/tftp.rb#L37-L47

I would not include it here because the TFTP step already has all the logic and done fairly early. Everything is rolled back anyway if that fails. Effectively reducing it to:
```suggestion
    templates = [provisioning_template]
```